### PR TITLE
Strided masked normalize.

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #68738
* __->__ #68694

Differential Revision: [D32724552](https://our.internmc.facebook.com/intern/diff/D32724552)

### CODE REVIEW

**Comment by cpuhrsch** [comment]
First question, [normalize](https://pytorch.org/docs/master/generated/torch.nn.functional.normalize.html?highlight=normalize#torch.nn.functional.normalize) is currently in torch.nn.functional. Should we make masked_normalize a functional as well?

**Comment by pearu** [comment]
> First question, [normalize](https://pytorch.org/docs/master/generated/torch.nn.functional.normalize.html?highlight=normalize#torch.nn.functional.normalize) is currently in torch.nn.functional. Should we make masked_normalize a functional as well?

It is not clear to me what makes a function `foo` a functional? Is it about making it available as `torch.nn.functional._masked.foo`? If so, the same question applies to other masked functions such as softmax, log_softmax, etc and I think the suggestion should be applied in some follow-up PR, not here.

File: test/test_masked.py [hunk 1]
```diff
@@ -126,28 +126,35 @@ def apply_masked_reduction_along_dim(op, input, *args, **kwargs):
     return output
 
 
-def apply_masked_normalization_along_dim(op, x, dim, dtype=None, mask=None):
```
Comment (cpuhrsch) [inline test/test_masked.py:129]: Based on this change, this PR might need rebasing now that https://github.com/pytorch/pytorch/pull/68584 is merged?

Comment (pearu) [reply test/test_masked.py:129]: Rebase done.

**Review by cpuhrsch — APPROVED** [review]

**Comment by cpuhrsch** [comment]
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32724552).
